pkg/upgrade/main.go

```diff
@@ -1,17 +1,27 @@
 package upgrade
 
 import (
+	"context"
 	"sort"
 
 	"github.com/Masterminds/semver"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	v1 "github.com/jaegertracing/jaeger-operator/pkg/apis/jaegertracing/v1"
+)
+
+type upgradeFunction = func(ctx context.Context, client client.Client, jaeger v1.Jaeger) (v1.Jaeger, error)
```

Comment:
Not sure these should have been moved into the main? They seemed fine where they were.